### PR TITLE
Bump Jaeger dependency on badger to 1.6.0 and resolve breaking changes

### PR DESCRIPTION
**Pull request by andreynkravtsov**
## Which problem is this PR solving?
- Need to upgrade badger dependency to 1.6.0

## Short description of the changes
- DefaultOptions is now a function with path default passed in
- Iterator now seems unhappy when you do it.Item() != null and expects to be checked against .Valid()

### CODE REVIEW

**Comment by andreynkravtsov** [comment]
Ooops, will revert the elastic ver bump

File: Gopkg.toml [hunk 1]
```diff
@@ -137,7 +137,7 @@ required = [
 
 [[constraint]]
   name = "github.com/dgraph-io/badger"
-  version = "=1.5.3"
+  version = "^1.6.0"
```
Comment (yurishkuro) [inline Gopkg.toml]: Given breaking changes, are you sure we want ^ and not =?

Comment (andreynkravtsov) [reply Gopkg.toml]: Good call, will fix

File: plugin/storage/badger/factory.go [hunk 2]
```diff
@@ -89,16 +88,14 @@ func (f *Factory) InitFromViper(v *viper.Viper) {
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.logger = logger
 
-	opts := badger.DefaultOptions
-	opts.TableLoadingMode = options.MemoryMap
```
[the remark this answers is not in the document]
Comment (andreynkravtsov) [reply plugin/storage/badger/factory.go]: I should have mentioned. In 1.6.0 TableLoadingMode is defaulted to MemoryMap

**Comment by burmanm** [comment]
Yeah, 1.6.0 is slower and has couple of new bugs I didn't want. So I would skip this version (that's why I've never sent a PR).

**Pull request closed without merging**